src/event.rs

```diff
@@ -221,6 +221,9 @@ impl<N: Notify> Processor<N> {
         ref_test: bool,
         size_info: SizeInfo,
     ) -> Processor<N> {
+        let hide_when_typing = config
+            .hide_cursor_when_typing
+            .unwrap_or(config.cursor().hide_when_typing);
```

Comment:
I slightly better deprecation interface might clean this up, but I'm not sure what that might look like atm.

Reply:
I've thought about that too. Potentially accessing the field directly through the config with a function might be a solution (like `config.hide_cursor_when_typing()`). That would keep all the deprecation handling in the `config.rs`.

However there might be a way to handle deprecations globally that can be used in the future too. It would be nice if it could be done while the config is loaded too. I haven't thought about it too much yet though.